app/instance-initializers/fastboot/ember-data-fastboot.js

```diff
@@ -2,13 +2,24 @@ export function initialize(applicationInstance) {
   let store = applicationInstance.lookup('service:store');
   let shoebox = applicationInstance.lookup('service:fastboot').get('shoebox');
 
+  let shoeboxSerializer = applicationInstance.lookup('serializer:-shoebox');
+  shoeboxSerializer.set('store', store);
+
   shoebox.put('ember-data-store', {
     get records() {
       return Object.keys(store.typeMaps).map(k => {
         let name = store.typeMaps[k].type.modelName;
         return store.peekAll(name).toArray();
       }).reduce((a,b) => a.concat(b), [])
-        .map(record => record.serialize({ includeId: true}))
+        .filter(record => {
+          // Filter records without an id because they cannot be deserialized.
+          // Call deleteRecord to remove it from any relationships.
+          if (record.get('id')) return true;
+          store.deleteRecord(record);
```

Comment:
This stuff is concerning, and suggests to me that we really don't want to be using a `DS.Serializer` implementation at all. Our needs are different, we just want to serialize the in-memory state of the store, and we control both sides.